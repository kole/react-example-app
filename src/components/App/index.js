import React from 'react';
import { observer, Provider } from 'mobx-react';
import styled from 'styled-components';

import Routes from '~/Routes';
import RootStore from '~/stores/RootStore';
const store = new RootStore();

const AppContainer = styled.div`
    display: flex;
    flex: 1;
    flex-direction: column;
    height: 100%;
`

@observer
export default class App extends React.Component {
    render() {
        return (
            <Provider rootStore={store}>
                <AppContainer>
                    <Routes />
                </AppContainer>
            </Provider>
        );
    }
}
