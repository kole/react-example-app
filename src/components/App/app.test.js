/* eslint-env jest */

import React from 'react';
import Shallow from 'react-test-renderer/shallow';
import App from './index.js';

describe('App', () => {
    test('Snapshot', () => {
        const renderer = new Shallow();
        renderer.render(<App />);
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });
});
