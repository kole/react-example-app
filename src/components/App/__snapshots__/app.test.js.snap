// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
<Provider
  rootStore={
    RootStoreClass {
      "AnswersStore": answersStoreClass {
        "RootStore": [Circular],
      },
      "QuestionsStore": questionsStoreClass {
        "RootStore": [Circular],
      },
      "ResultsStore": resultsStoreClass {
        "RootStore": [Circular],
      },
    }
  }
>
  <styled.div>
    <Routes />
  </styled.div>
</Provider>
`;
